src/KafkaFlow/Producers/IMessageProducer.cs

```diff
@@ -62,6 +62,7 @@ public interface IMessageProducer
     /// <param name="headers">The message headers</param>
     /// <param name="deliveryHandler">A handler with the operation result</param>
     /// <param name="partition">The partition where the message will be produced, if no partition is provided it will be calculated using the message key</param>
+    [Obsolete("This method will be remove in the next major release, please use ProduceAsync() instead")]
```

Comment:
Why are we marking the Produce methods as obsolete? In scenarios where the message throughput is high, the non async producer can produce at a significant higher rate since it doesn't need to wait for the report.

Why can't the framework support this?